crates/ecs_bench_suite/src/recs/simple_iter.rs

```diff
@@ -0,0 +1,71 @@
+use cgmath::*;
+use ecs::systems::{IntoSystem, Read, System, Write};
+use ecs::{
+    Application, ApplicationBuilder, BasicApplication, BasicApplicationBuilder, Executor, Schedule,
+    Sequential,
+};
+use scheduler::schedule::PrecedenceGraph;
+use std::sync::{Mutex, OnceLock};
+
+#[derive(Copy, Clone, Debug)]
+struct Transform(Matrix4<f32>);
+
+#[derive(Copy, Clone, Debug)]
+struct Position(Vector3<f32>);
+
+#[derive(Copy, Clone, Debug)]
+struct Rotation(Vector3<f32>);
+
+#[derive(Copy, Clone, Debug)]
+struct Velocity(Vector3<f32>);
+
+static APPLICATION: OnceLock<BasicApplication> = OnceLock::new();
+static SCHEDULE: Mutex<Option<PrecedenceGraph>> = Mutex::new(None);
+static SYSTEMS: OnceLock<Vec<Box<dyn System>>> = OnceLock::new();
+
+pub struct Benchmark;
+
+impl Benchmark {
+    pub fn new() -> Self {
+        SYSTEMS.get_or_init(|| {
+            let movement_system: Box<dyn System> = Box::new(movement_system.into_system());
+            vec![movement_system]
+        });
+
+        APPLICATION.get_or_init(|| {
+            let mut app = BasicApplicationBuilder::default().build();
+
+            for _ in 0..10_000 {
+                let entity = app.create_entity().unwrap();
+                app.add_component(entity, Matrix4::from_scale(1.0)).unwrap();
+                app.add_component(entity, Position(Vector3::unit_x()))
+                    .unwrap();
+                app.add_component(entity, Rotation(Vector3::unit_x()))
+                    .unwrap();
+                app.add_component(entity, Velocity(Vector3::unit_x()))
+                    .unwrap();
+            }
+
+            let systems = SYSTEMS.get().unwrap();
+            let schedule = PrecedenceGraph::generate(systems).unwrap();
+            let mut schedule_guard = SCHEDULE.lock().unwrap();
+            *schedule_guard = Some(schedule);
+
+            app
+        });
+
+        Self
+    }
+
+    pub fn run(&mut self) {
+        let app = APPLICATION.get().unwrap();
+        let mut schedule_guard = SCHEDULE.lock().unwrap();
+        let schedule = schedule_guard.as_mut().unwrap();
+
+        Sequential.execute_once(schedule, &app.world).unwrap();
```

Comment:
Very good point! I've fixed this in commit 2bc53bc3588733051e8e47cbdeb42f8bb5c38bec.

Reply:
`frag_iter` bench went from `7.5±0.15µs` to `6.2±0.01µs` so that's good!

Reply:
Nice! Is it possible to the same in this bench or would it go against the specifications set by the bench suite? Referring to "This benchmark is designed to test the _core overheads_ involved in component iteration in best-case conditions."

Reply:
Oh I totally missed that this comment was for `simple_iter` since I saw the same code in `frag_iter`. I'll do the same for this one. 

Since Bevy gets to skip their entire scheduler and use a Query directly, we should be allowed to as well.

Reply:
I've fixed this in commit 02aa4b4278bbd17602e75aa7be9646ea2a0bce05, if you'd like to take a look to make sure it seems reasonable.